src/tiles/internal.tsx

```diff
@@ -77,7 +77,9 @@ export default function InternalTiles({
                 )}
                 key={item.value}
                 data-value={item.value}
-                onClick={() => item.value !== value && fireNonCancelableEvent(onChange, { value: item.value })}
+                onClick={() =>
+                  !item.disabled && item.value !== value && fireNonCancelableEvent(onChange, { value: item.value })
```

Comment:
I think it would be a good idea to rewrite the short-circuit logic as an if-block, and maybe extract this into a proper function; that would improve the readability of the code a lot imo.

Reply:
Thanks! I've integrated this suggestion in  my other CR for the AbstractSwitch: https://github.com/cloudscape-design/components/pull/388
I will close this PR in favour of the other (larger) one.